tests/unit_tests/dft/include/compute_inplace.hpp

```diff
@@ -0,0 +1,157 @@
+
+    const size_t container_size = domain == dft::domain::REAL ? conjugate_even_size : size;
+
+    sycl::buffer<InputType, 1> inout_dev{ sycl::range<1>(container_size) };
+    std::vector<InputType> out_host(container_size);
+
+    copy_to_device(sycl_queue, input, inout_dev);
+
+    commit_descriptor(descriptor, sycl_queue);
+
+    try {
+        dft::compute_forward<std::remove_reference_t<decltype(descriptor)>, InputType>(descriptor,
+                                                                                       inout_dev);
+    }
+    catch (oneapi::mkl::unimplemented &e) {
+        std::cout << "Skipping test because: \"" << e.what() << "\"" << std::endl;
+        return test_skipped;
+    }
+
+    copy_to_host(sycl_queue, inout_dev, out_host);
+
+    if constexpr (domain == dft::domain::REAL) {
+        std::vector<InputType> out_host_ref_conjugate = std::vector<InputType>(conjugate_even_size);
+        for (int i = 0; i < out_host_ref_conjugate.size(); i += 2) {
+            out_host_ref_conjugate[i] = out_host_ref[i / 2].real();
+            out_host_ref_conjugate[i + 1] = out_host_ref[i / 2].imag();
+        }
+        EXPECT_TRUE(check_equal_vector(out_host.data(), out_host_ref_conjugate.data(),
+                                       out_host.size(), 1, error_margin, std::cout));
+    }
+    else {
+        EXPECT_TRUE(check_equal_vector(out_host.data(), out_host_ref.data(), out_host.size(), 1,
+                                       error_margin, std::cout));
+    }
+
+    descriptor_t descriptor_back{ size };
+    descriptor_back.set_value(oneapi::mkl::dft::config_param::PLACEMENT,
+                              oneapi::mkl::dft::config_value::INPLACE);
+    descriptor_back.set_value(dft::config_param::BACKWARD_SCALE, (1.0 / size));
+    commit_descriptor(descriptor_back, sycl_queue);
+
+    try {
+        dft::compute_backward<std::remove_reference_t<decltype(descriptor_back)>, InputType>(
+            descriptor_back, inout_dev);
+    }
+    catch (oneapi::mkl::unimplemented &e) {
+        std::cout << "Skipping test because: \"" << e.what() << "\"" << std::endl;
+        return test_skipped;
+    }
+
+    copy_to_host(sycl_queue, inout_dev, out_host);
+
+    EXPECT_TRUE(check_equal_vector(out_host.data(), input.data(), input.size(), 1, error_margin,
+                                   std::cout));
+
+    return !::testing::Test::HasFailure();
+}
+
+template <dft::precision precision, dft::domain domain>
+int DFT_Test<precision, domain>::test_in_place_USM() {
+    if (!init()) {
+        return test_skipped;
+    }
+
+    descriptor_t descriptor{ size };
+    descriptor.set_value(oneapi::mkl::dft::config_param::PLACEMENT,
+                         oneapi::mkl::dft::config_value::INPLACE);
+    commit_descriptor(descriptor, sycl_queue);
+
+    const size_t container_size = domain == dft::domain::REAL ? conjugate_even_size : size;
+
+    auto ua_input = usm_allocator<InputType, usm::alloc::shared, 64>(cxt, *dev);
```

Comment:
Might be worth checking that the device supports `aspect::usm_shared_allocations`.